### ISSUE TITLE
docs zip contains unwanted package source files

### ISSUE THREAD

**Issue by eachimei** [devpi/devpi#1002]
- [x] Are you using the latest released version?
      If something is preventing you from trying with the newest version,
      we would like to know so we can make it easier.

Yes: devpi-client==6.0.5

- [x] Provide the output of ``pip list`` from the virtual environment you are using.
```
devpi-client                             6.0.5
devpi-common                             3.7.2
```
- [x] Provide the Python and operating system versions under which the issue occurs.

Windows 10, ,Python 3.10 and also Python 3.8

- [x] If possible, provide a minimal example to reproduce the issue.

**Steps to reproduce:**
1. Create a simple package (mine is using pyproject.toml).
2. Run:
`devpi upload --wheel --with-docs`
For simplicity, you can also add `--dry-run`:
`devpi upload --wheel --with-docs --dry-run`
3. Then, inspect the generated `.doc.zip` file.

**Problem statement:**
In short: the docs file (doc.zip) contains by mistake unwanted package source files.
The root-cause:
Seems that when the wheel is being generated, a directory named `build` is created and used by the build-frontend package (e.g. `python -m build --wheel`). This directory is not cleaned up between the phase when `devpi` creates the wheel and the phase `devpi` calls `sphinx` to create the docs. The docs phase uses the same `build` directory. As a result, the generated `.doc.zip` contains not just sphinx output files, but also the intermediate wheel sources. For example, the zip contains a directory named `lib` that contains all the package sources and also a directory named `bdist.win-amd64` (since my example was running on Windows).

**Comment by eachimei** [comment]
Submitted fix in PR https://github.com/devpi/devpi/pull/1003

**Comment by fschulze** [comment]
Can you try the following dev release?
https://m.devpi.net/fschulze/dev/devpi-client/6.0.6.dev0
https://m.devpi.net/fschulze/dev/+f/f77/7d6454662e3ac/devpi_client-6.0.6.dev0-py2.py3-none-any.whl

I added a test and a news snippet in https://github.com/fschulze/devpi/commit/53d9df2fc2acc0862d091d3b2f7d10a21bcd91c7

**Comment by eachimei** [comment]
Tested 6.0.6dev0 -- looks good: no `wheel` artifacts inside the `doc.zip`!
